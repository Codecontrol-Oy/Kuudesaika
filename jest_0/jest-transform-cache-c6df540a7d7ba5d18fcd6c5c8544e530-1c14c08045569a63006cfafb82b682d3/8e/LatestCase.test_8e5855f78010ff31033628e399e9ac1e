'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _LatestCase = require('./LatestCase');var _LatestCase2 = _interopRequireDefault(_LatestCase);
var _Label = require('grommet/components/Label');var _Label2 = _interopRequireDefault(_Label);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe("<LatestCase />", function () {
    it("Matches snapshot", function () {
        var latestCase = shallow(
        _react2.default.createElement(_LatestCase2.default, { url: "http://testurl.com",
            id: 2,
            data_source: "testSource",
            actions: null,
            geometries: null,
            created_at: "2017-01-1T10:00:00.00000Z",
            modified_at: "2017-01-1T10:00:00.00000Z",
            origin_id: "1",
            title: "Test case",
            register_id: "CASE 12345-1",
            'function': null,
            attachments: null }));


        expect(latestCase).toMatchSnapshot();
    });

    it('should contain correct elements', function () {
        var latestCase = shallow(_react2.default.createElement(_LatestCase2.default, { url: "http://testurl.com",
            id: 2,
            data_source: "testSource",
            actions: null,
            geometries: null,
            created_at: "2017-01-1T10:00:00.00000Z",
            modified_at: "2017-01-1T10:00:00.00000Z",
            origin_id: "1",
            title: "Test case",
            register_id: "CASE 12345-1",
            'function': null,
            attachments: null }));
        expect(latestCase.find('Card').length).toEqual(1);
    });

    it('Latest case should contain correct values', function () {
        var latestCase = shallow(_react2.default.createElement(_LatestCase2.default, { url: "http://testurl.com",
            id: 2,
            data_source: "testSource",
            actions: null,
            geometries: null,
            created_at: "2017-01-1T10:00:00.00000Z",
            modified_at: "2017-01-1T10:00:00.00000Z",
            origin_id: "1",
            title: "Test case",
            register_id: "CASE 12345-1",
            'function': null,
            attachments: null,
            subject: "Test subject" }));
        expect(latestCase.find('Card').prop("label")).toEqual("CASE 12345-1");
        expect(latestCase.find('Card').props().description).toEqual(_react2.default.createElement(_Label2.default, { announce: true }, 'Test case'));
        expect(latestCase.find('Card').props().heading).toEqual("Test subject");
    });

});