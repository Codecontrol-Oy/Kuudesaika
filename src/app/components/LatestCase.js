import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from 'grommet/components/Card';
import Label from 'grommet/components/Label';
import Anchor from 'grommet/components/Anchor';
import Timestamp from 'grommet/components/Timestamp';
import axios from 'axios';
export default class LatestCase extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
        action: null
    }
  }

  componentDidMount = () => {
  }

  render () {
    if(this.state.action) {

    }
    return (
        <div>
            <Card contentPad={"medium"} 
                  style={{borderBottom: "2px solid #CCC"}}
                  label={<Timestamp value={this.props.created_at} fields={"date"}/>}
                  heading={this.props.subject}
                  description={<Label announce={true}>{this.props.title}</Label>}/>
        </div>
    );
  }
}

LatestCase.propTypes = {
    url: PropTypes.string,
    id: PropTypes.number,
    data_source: PropTypes.string,
    actions: PropTypes.array,
    geometries: PropTypes.array,
    created_at: PropTypes.string,
    modified_at: PropTypes.string,
    origin_id: PropTypes.string,
    title: PropTypes.string,
    register_id: PropTypes.string,
    function: PropTypes.string,
    attachments: PropTypes.array,
    subject: PropTypes.string
};