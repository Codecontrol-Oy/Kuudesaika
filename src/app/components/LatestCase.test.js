import React from 'react';
import LatestCase from './LatestCase';
import Label from 'grommet/components/Label';

describe("<LatestCase />", () => {
    it("Matches snapshot", () => {
    const latestCase = shallow(
        <LatestCase url={"http://testurl.com"}
                    id={2}
                    data_source={"testSource"}
                    actions={null}
                    geometries={null}
                    created_at={"2017-01-1T10:00:00.00000Z"}
                    modified_at={"2017-01-1T10:00:00.00000Z"}
                    origin_id={"1"}
                    title={"Test case"}
                    register_id={"CASE 12345-1"}
                    function={null}
                    attachments={null} />
    );

    expect(latestCase).toMatchSnapshot();
    });

    it('should contain correct elements', () => {
        const latestCase = shallow(<LatestCase url={"http://testurl.com"}
                    id={2}
                    data_source={"testSource"}
                    actions={null}
                    geometries={null}
                    created_at={"2017-01-1T10:00:00.00000Z"}
                    modified_at={"2017-01-1T10:00:00.00000Z"}
                    origin_id={"1"}
                    title={"Test case"}
                    register_id={"CASE 12345-1"}
                    function={null}
                    attachments={null} />);
        expect(latestCase.find('Card').length).toEqual(1);
    });

    it('Latest case should contain correct values', () => {
        const latestCase = shallow(<LatestCase url={"http://testurl.com"}
                    id={2}
                    data_source={"testSource"}
                    actions={null}
                    geometries={null}
                    created_at={"2017-01-1T10:00:00.00000Z"}
                    modified_at={"2017-01-1T10:00:00.00000Z"}
                    origin_id={"1"}
                    title={"Test case"}
                    register_id={"CASE 12345-1"}
                    function={null}
                    attachments={null}
                    subject={"Test subject"} />);
        expect(latestCase.find('Card').prop("label")).toEqual("CASE 12345-1");
        expect(latestCase.find('Card').props().description).toEqual(<Label>Test case</Label>);
        expect(latestCase.find('Card').props().heading).toEqual("Test subject");
    });

})
