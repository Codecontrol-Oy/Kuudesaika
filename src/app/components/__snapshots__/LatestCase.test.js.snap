// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LatestCase /> Matches snapshot 1`] = `
<div>
  <Card
    a11yTitle="Card"
    contentPad="medium"
    description={
      <Label
        announce={true}
        size="medium"
      >
        Test case
      </Label>
    }
    headingStrong={true}
    label="CASE 12345-1"
    link={
      <Anchor
        animateIcon={true}
        method="push"
        onClick={[Function]}
        tag="a"
      >
        Siirry
      </Anchor>
    }
    style={
      Object {
        "borderBottom": "2px solid #CCC",
      }
    }
    textSize="small"
  />
</div>
`;
